ui/src/app/ui-banner/ui-banner.tsx

```diff
@@ -0,0 +1,49 @@
+import * as React from 'react';
+require('./ui-banner.scss');
+
+export interface UiBannerProps extends React.Props<any> {
+    items: {apiUrl:string, fields: Array<string> };
+}
+
+
+export class Banner extends React.Component<UiBannerProps>{
```

Comment:
Please use `DataLoader` component to load data asynchronously. `DataLoader` takes care of rendering `loading`  and `failed` state

```
import {DataLoader} from '../../../shared/components';

export const Banner = () => (
   <DataLoader load={() => fetch(this.props.items.apiUrl)}>
    {(apiData) => (
        <div className='ui_banner' style={{visibility: this.state.bannerVisible? "visible": "hidden"}} >
                <button className='ui_banner__close' aria-hidden='true' onClick={this.setBanner}>
                    <span>
                        <i className='argo-icon-close' aria-hidden='true'/>
                    </span>
                </button>
                <div className="ui_banner__text">
                    ... # render apiData
                </div>
            </div>
    )}
    </DataLoader>
);
```